### DISCUSSION TITLE
Extensiones en Visual Studio Code

### DISCUSSION THREAD

**Discussion by JorgeGarcia9006** [LaunchX-InnovaccionVirtual/FrontEnd-Mision#158] [Q&A]
Hola a todos, una pregunata rapida, ¿sabran que extenciones serian utiles para usar en visualstudiocode la practica del pastel?

**Comment by Ingrzs** [comment]
Pues para html yo uso, auto rename tag, html wrapper y open in browser , ahi muchas es cuestión de gustos e utilidad.

**Comment by avexyk** [comment]
Pues, me parece que el visual studio tiene autocompletado, no se si este activado por defecto. Me refiero a algo así.

![img18](https://user-images.githubusercontent.com/32181532/155910029-bf0aac35-1a3f-4ce4-9651-7ad011bb8081.png)

En ese caso, creo que ese autocompletado te ayudará a agilizar tu trabajo. Por otro lado. Te recomendaría estas extensiones:

![img16](https://user-images.githubusercontent.com/32181532/155910111-74bea8c2-5a24-4328-a6c3-869a5ef8e351.png)

Esa hace que tu directorio de carpetas y archivos se vea así por ejemplo, es más identificable:

![img18](https://user-images.githubusercontent.com/32181532/155910164-dc275410-8094-410a-815a-6ce0f1bcfb05.png)

Otra sería esta:

![img15](https://user-images.githubusercontent.com/32181532/155910211-52d7ac82-6757-4365-88ca-de6139099607.png)

Te crea un pequeño botón que puedes usar para abrir un servidor local en el navegador usando el puerto 5500, se abre por defecto tú navegador predeterminado.

![img18](https://user-images.githubusercontent.com/32181532/155910292-67a706b6-88e8-4b11-bbf5-dc0e8d75db78.png)

Presionas el que dice Go Live y listo. Verás lo que llevas trabajando y tus cambio se reflejaran en vivo mientras vayas guardando.

Otra más:

[Emmet Documentation](https://docs.emmet.io/cheat-sheet/)

Ese es un recurso de abreviaciones que te puede ser útil para agilizar aún más tu escritura, no sé si todas son válidas pero puedes ir probando por tu cuenta en Visual Studio Code. Por ejemplo:

![img19](https://user-images.githubusercontent.com/32181532/155910565-6a82c4cb-dc9d-4d85-b152-5c316be974e8.png)

Daría como resultado:

![img20](https://user-images.githubusercontent.com/32181532/155910590-ba97076d-446a-4b5e-9442-c3397b16be05.png)

Esta otra sería cuando toquemos JavaScript, para ver el cierre de las llaves en caso de que existan varias operaciones dentro de una función, condición o algo similar.

![img12](https://user-images.githubusercontent.com/32181532/155911309-d45a53b6-48e4-4ff5-9bf2-eaafd871037e.png)

Algo así:

![img16](https://user-images.githubusercontent.com/32181532/155911452-6739ab57-cd8a-44a6-8cdd-72d0c3175b2c.png)

Si notas se puede identificar a simple vista a quien pertenece la apertura y cierre de llaves.

También puedes revisar esto, que son un compilado de los atajos que tiene Visual Studio Code. Te pueden ser útiles.

[Atajos de VSCode](https://code.visualstudio.com/shortcuts/keyboard-shortcuts-windows.pdf)

Un pequeño aporte.

**Reply by Ivanflx19** [reply]
Muchas gracias hermano, son muy utiles tus aportes ✌

**Reply by JorgeEHE** [reply]
Excelente aporte, la parte visual siempre es muy útil y poder ver la organización de tus proyectos en forma gráfica ayuda bastante

**Comment by doguedogue** [comment]
Prettier https://marketplace.visualstudio.com/items?itemName=esbenp.prettier-vscode

Aquí otras para un web developer https://codeforgeek.com/best-visual-studio-code-extensions-web-development/
